substrate/client/service/src/builder.rs

```diff
@@ -834,17 +834,17 @@ where
 	}
 
 	// create transactions protocol and add it to the list of supported protocols of
-	// `network_params`
-	let transactions_handler_proto = sc_network_transactions::TransactionsHandlerPrototype::new(
-		protocol_id.clone(),
-		client
-			.block_hash(0u32.into())
-			.ok()
-			.flatten()
-			.expect("Genesis block exists; qed"),
-		config.chain_spec.fork_id(),
-	);
-	net_config.add_notification_protocol(transactions_handler_proto.set_config());
+	let (transactions_handler_proto, transactions_config) =
+		sc_network_transactions::TransactionsHandlerPrototype::new(
+			protocol_id.clone(),
+			client
+				.block_hash(0u32.into())
```

Comment:
small nit: Could we place this in a variable `let genesis_hash = ..`? Such that it is shared between `warp_sync_params` and `/transactions/1`?